launch/test/launch/test_logging.py

```diff
@@ -249,3 +250,56 @@ def emit(self, record):
     assert path in outputs
     assert len(outputs[path]) == 1
     assert outputs[path][0].endswith('baz')
+
+
+def fake_make_unique_log_dir(*, base_path):
+    # Passthrough; do not create the directory
+    return base_path
+
+
+@mock.patch('launch.logging._make_unique_log_dir', mock.MagicMock(wraps=fake_make_unique_log_dir))
+def test_get_logging_directory():
+    launch.logging.launch_config.reset()
+    os.environ.pop('ROS2_LOG_DIR', None)
+    os.environ.pop('ROS2_HOME', None)
+    home = os.environ.get('HOME', None)
+    assert home
+
+    # Default case without ROS2_LOG_DIR or ROS2_HOME being set (but with HOME)
+    default_dir = os.path.join(home, '.ros/log')
+    launch.logging.launch_config.log_dir = None
+    assert launch.logging.launch_config.log_dir == default_dir
```

Comment:
nit^N: add a comment (the first time) explaining that setting `launch.logging.launch_config.log_dir = None` will ensure that the launch config checks the environment again.

This is super confusing, I needed to check the code to understand what was happening :joy:.

Reply:
:laughing: sorry! Added in 46e727df92a4b4a2914e7a5863bb9fb51eeb04d4